### PR TITLE
Feature: select custom CA certificate to allow using self-signed certificates

### PR DESCRIPTION
**Pull request by danibix95**
## Proposal

Added new flag `--ca-cert` to the root command to allow selecting a custom CA certificate when connecting to the console server.  
This feature can be useful for dealing with cli connection with servers whose certificate are self-signed.

This PR must be merged only after #43. 

E.g.

    miactl --ca-cert=./ca-cert.pem console deploy

### CODE REVIEW

File: cmd/console/deploy/deploy_test.go [hunk 1]
```diff
@@ -136,6 +139,62 @@ func TestNewDeployCmd(t *testing.T) {
 		require.Equal(t, expectedRow, tableRows[1])
 	})
 
+	t.Run("successful deploy - select custom CA certificate", func(t *testing.T) {
```
Comment (davidebianchi) [inline cmd/console/deploy/deploy_test.go:142]: Can we test that calling a server with a wrong ca certs, it returns error?

File: cmd/console/deploy/status_test.go [hunk 2]
```diff
@@ -172,6 +175,54 @@ func TestNewStatusCmd(t *testing.T) {
 		require.Equal(t, expectedRow, tableRows[1])
 	})
 
+	t.Run("get pipeline status with success - select custom CA certificate", func(t *testing.T) {
```
Comment (davidebianchi) [inline cmd/console/deploy/status_test.go:178]: Can we test that calling a server with a wrong ca certs, it returns error?

File: cmd/login/login_test.go [hunk 3]
```diff
@@ -103,6 +106,49 @@ func TestNewLoginCmd(t *testing.T) {
 		require.Equal(t, expectedAccessToken, accessToken, "Access token differs from expected")
 	})
 
+	t.Run("successful login - select custom CA certificate", func(t *testing.T) {
```
Comment (davidebianchi) [inline cmd/login/login_test.go:109]: Can we test that calling a server with a wrong ca certs, it returns error?

File: sdk/auth/auth_test.go [hunk 4]
```diff
@@ -73,13 +75,49 @@ func TestLogin(t *testing.T) {
...
 
-		insecureClient := getTestClient(t, s.URL, true)
+		insecureClient := getTestClient(t, s.URL, true, nil)
 		accessToken, err := insecureClient.Login(username, password, providerID)
 
 		require.Nil(t, err)
 		require.Equal(t, expectedAccessToken, accessToken, "Access token differs from expected")
 	})
 
+	t.Run("successful login - use custom certificate", func(t *testing.T) {
```
Comment (davidebianchi) [inline sdk/auth/auth_test.go:85]: Can we test that calling a server with a wrong ca certs, it returns error?

File: cmd/console/deploy/deploy.go [hunk 5]
```diff
@@ -41,25 +42,26 @@ func NewDeployCmd() *cobra.Command {
...
 			})
 			if err != nil {
 				return err
 			}
 
 			deployData, err := f.MiaClient.Deploy.Trigger(projectId, cfg)
 			if err != nil {
-				f.Renderer.Error(err).Render()
-				return nil
+				return err
```
Comment (davidebianchi) [inline cmd/console/deploy/deploy.go:64]: Why it is not used the renderer?

File: cmd/console/deploy/status.go [hunk 6]
```diff
@@ -66,8 +69,7 @@ func NewStatusCmd() *cobra.Command {
 
 			result, err := f.MiaClient.Deploy.GetDeployStatus(projectId, pipelineId, environment)
 			if err != nil {
-				f.Renderer.Error(err).Render()
-				return nil
+				return err
```
Comment (davidebianchi) [inline cmd/console/deploy/status.go:72]: Also here

File: cmd/completion_test.go [hunk 7]
```diff
@@ -8,16 +8,14 @@ import (
 
 func TestCompletion(t *testing.T) {
 	t.Run("without correct args", func(t *testing.T) {
-		out, err := executeCommand(NewRootCmd(), "completion", "not-correct-arg")
+		_, err := executeCommand(NewRootCmd(), "completion", "not-correct-arg")
```
Comment (davidebianchi) [inline cmd/completion_test.go:11]: Why this change? outstd will not show the error?

File: cmd/console/deploy/deploy.go (hunk 5 above)
Comment (danibix95) [reply cmd/console/deploy/deploy.go:64]: Because using just the renderer prevents the command to exit with a status code different from 0.

The error is then printed to std out by these lines: https://github.com/mia-platform/miactl/blob/feature/select-custom-certificate/cmd/root.go#L53-L56 which are in charge also to select correct status code.

**Note:** returning an error is not a solution since the error message would be printed three times:
- from the renderer
- from cobra error handler (disabled setting `SilenceErrors: true` in root command config)
- from miactl custom error handler

File: cmd/completion_test.go (hunk 7 above)
Comment (danibix95) [reply cmd/completion_test.go:11]: This change is due to the enabling of the SilenceErrors property in the root command (https://github.com/mia-platform/miactl/blob/feature/select-custom-certificate/cmd/root.go#L32-L35).  
As a result, the error is not printed to std out automatically by cobra when the `executeCommand` function is run on selected command, requiring the test to be changed to just verify error content.

Nonetheless, due to these lines https://github.com/mia-platform/miactl/blob/feature/select-custom-certificate/cmd/root.go#L53-L56, the behaviour is retained.
